internal/pkg/http/client/pool_test.go

```diff
@@ -221,9 +221,11 @@ func TestErrorInSubRequest_ParallelRequests(t *testing.T) {
 
 	// Register callbacks
 	request1.OnResponse(func(response *Response) {
+		time.Sleep(1 * time.Millisecond)
 		logger.Info("Request1 processed.")
 	})
 	request2.OnResponse(func(response *Response) {
+		time.Sleep(1 * time.Millisecond)
 		logger.Info("Request2 processed.")
 	})
```

Comment:
Prisiel som na to preco tento test miekedy pada:
![image](https://user-images.githubusercontent.com/19371734/162378785-6fcd16b8-a3c1-4352-99d9-d75b052e2894.png)


- Testuju sa tu 2 paralelne requesty.
- request1 je OK
- request2 je ERR
- `request1.WaitFor(request2)`, tj. response z req1 sa ma spracovat az po req2
- ak nejaky request v `pool` spadne, tak sa cely `pool` hned ukoncuje, a beziace requesty sa ukoncia tiez hned.
- preto ak request2 kompletne dobehne skor, ... a request1 sa este nacitava z HTTP, tak sa request1 ukonci s `context canceled`, co je validne ... pool nevrati tuto chybu, ale chybu z `request2`, ktora bola pricina, a ukoncila pool, je na to `assert.Contains`, riadok `242`.

Preto som pridal `time.Sleep(1 * time.Millisecond)`, a teda umelo natiahnem oba requesty.
Tym padom urcite obidva stiahnu data z HTTP, ... a testuje sa iba poradie callbackov a spracovanie chyby (co riesi tento test).